x/leverage/types/weighted_coin.go

```diff
@@ -135,6 +152,7 @@ func (wdc WeightedDecCoins) Total(denom string) sdk.Dec {
 }
 
 // Sub subtracts a sdk.DecCoin from a WeightedDecCoins. Panics if the result would be negative.
+// Does not sort or change the order of the input denoms, combine duplicates, or remove zero coins.
```

Comment:
I think there are cases where this would have been a significant bug. Its fixes should be merged before release.